### PR TITLE
Modlog: Remove Rooms global in PM

### PR DESCRIPTION
**Pull request by mia-pi-git**

### CODE REVIEW

File: server/modlog.ts [hunk 1]
```diff
@@ -467,6 +469,7 @@ export class Modlog {
 // even though it's a type not a function...
 type ModlogResult = ModlogEntry | undefined;
 
+export const mainModlog = new Modlog(MODLOG_PATH, MODLOG_DB_PATH);
```
Comment (Zarel) [inline server/modlog.ts:472]: Does this API need to be changed?

Comment (Zarel) [reply server/modlog.ts:472]: Oh, I see, it's used by Modlog.

Comment (Zarel) [reply server/modlog.ts:472]: I would call it `modlog` lowercase `m` unless there's a reason otherwise?

Comment (Zarel) [reply server/modlog.ts:472]: I guess it conflicts with other variables called `modlog` in this file? This is fine.

**Comment by Zarel** [comment]
@AnnikaCodes `Rooms` is not designed to be imported outside of the main process, and importing it in Modlog subprocesses causes them to spawn unused `room-battle` sub-subprocesses. That's what this PR fixes. Can you review to make sure it works as you intended?

**Review by AnnikaCodes — APPROVED** [review]
These changes look fine. #7620 moves modlog entirely to `Chat`, which I think is a much better solution than `mainModlog` trickery, but if this PR will end up being faster to merge then I think it's a good interim solution. Thanks!

**Pull request merged**